src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationMockTests.cs

```diff
@@ -219,12 +215,12 @@ public void ConfiguringTheClient()
 
             var client = new ConfigurationClient(connectionString, options);
 
-            var e = Assert.ThrowsAsync<RequestFailedException>(async () =>
-            {
-                await client.GetAsync(key: s_testSetting.Key, options: null, CancellationToken.None);
-            });
-            var response = e.Response;
-            response.Dispose();
+            try {
+                ConfigurationSetting setting = await client.GetAsync(key: s_testSetting.Key, options: null, CancellationToken.None);
```

Comment:
Why did you switch from using Assert.ThrowsAsync to try catch?
This test would now pass if an exception is not thrown.

Reply:
I am a bit confused about what this test is supposed to test. Clearly it is not a good test to test that an exception is thrown when a wrong retry policy is selected. Anyway, I will try to clean it up a bit more.